src/platform/sdl/main.c

```diff
@@ -321,3 +338,63 @@ static void mSDLDeinit(struct mSDLRenderer* renderer) {
 
 	SDL_Quit();
 }
+
+static void _setLogger(struct mCore* core) {
+	int fakeBool = 0;
+	bool logToFile = false;
+
+	if (mCoreConfigGetIntValue(&core->config, "logToStdout", &fakeBool)) {
+		_logToStdout = fakeBool;
+	}
+	if (mCoreConfigGetIntValue(&core->config, "logToFile", &fakeBool)) {
+		logToFile = fakeBool;
+	}
+	const char* logFile = mCoreConfigGetValue(&core->config, "logFile");
+	
+	if (logToFile && logFile) {
+		_logFile = VFileOpen(logFile, O_WRONLY | O_CREAT | O_APPEND);
+	}
+
+	// Create the filter
+	mLogFilterInit(&_filter);
+	mLogFilterLoad(&_filter, &core->config);
+
+	// Fill the logger
+	_logger.log = _mCoreLog;
+	_logger.filter = &_filter;
+}
+
+static void _mCoreLog(struct mLogger* logger, int category, enum mLogLevel level, const char* format, va_list args) {
+	struct mCoreThread* thread = mCoreThreadGet();
+	if (thread && level == mLOG_FATAL) {
+		mCoreThreadMarkCrashed(thread);
+	}
+	
+	if (!mLogFilterTest(logger->filter, category, level)) {
+		return;
+	}
+
+	char buffer[MAX_LOG_BUF];
+
+	// Prepare the string
+	size_t length = snprintf(buffer, sizeof(buffer), "%s: ", mLogCategoryName(category));
+	if (length < sizeof(buffer)) {
+		length += vsnprintf(buffer + length, sizeof(buffer) - length, format, args);
+		if (length < sizeof(buffer)) {
+			length += snprintf(buffer + length, sizeof(buffer) - length, "\n");
+		}
+	}
```

Comment:
```suggestion
	}
	if (length < sizeof(buffer)) {
		length += snprintf(buffer + length, sizeof(buffer) - length, "\n");
	}
```
The can actually be separated since the latter one will never be true if the former one isn't.